smart-queue/enqueue.js

```diff
@@ -0,0 +1,64 @@
+import * as QueueActionTypes from './action-types';
+
+function validKey(key) {
+  if (key === null) {
+    const message = "Missing key, every queue action should have a key!";
+    console.log(message);
+    return false;
+  }
+  return true;
+}
+
+function indexOfItem(array, key) {
+  return array.findIndex(({meta:{offline:{queue}}}) => queue.key === key);
+}
+
+export function enqueue(array, item) {
+  const { meta: { offline : { queue: { method = 'UNKNOWN', key = null } } } } = item;
+  let index, existingItem;
+
+  if (!validKey(key)) {
+    return array;
```

Comment:
I think that we should allow the user to use non-smart actions just for flexibility's sake. I'm sorry I know I told you to ignore the actions if no key was provided but I'm not sure anymore.